### PR TITLE
HttpService (JSONEncode and JSONDecode)

### LINKED ISSUE #27: Introduce JSON library
To implement `HttpService:JSONDecode` and `HTTPService:JSONEncode`, we need to either bundle a Lua JSON library into Lemur's source, or depend on a new rock like dkjson for parsing.

### PR DESCRIPTION
**Pull request by Kampfkarren**
Closes #27

### CODE REVIEW

**Comment by Kampfkarren** [comment]
Luacov whyyy 😢

File: lib/instances/HttpService_spec.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
+if not pcall(require, "dkjson") then
+	return
+end
```
Comment (LPGhatguy) [inline lib/instances/HttpService_spec.lua]: We should require that `dkjson` is installed to run the test suite.

File: lib/instances/HttpService.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
+local BaseInstance = import("./BaseInstance")
+local exists, json = pcall(require, "dkjson")
+
+local HttpService = BaseInstance:extend("HttpService")
+
+function HttpService.prototype:JSONEncode(input)
+	if not exists then
+		error("JSONEncode is disabled while dkjson is not installed.")
```
Comment (LPGhatguy) [inline lib/instances/HttpService.lua]: We should change these methods to give the user an obvious action:

```
Please install dkjson to use HttpService:JSONEncode.
```

**Comment by Kampfkarren** [comment]
Completely forgot about Travis. I'll fix all of those when I get the chance.

**Comment by Kampfkarren** [comment]
Home now, will get started.

**Comment by Kampfkarren** [comment]
There's nothing I can do about the coverage, it's the same reason we can't test luafilesystem not being installed or whatever it was.

**Comment by LPGhatguy** [comment]
That's fine, there's no reason to be a stickler about this kind of coverage.

**Pull request merged**